datajunction-query/djqs/config.py

```diff
@@ -3,94 +3,202 @@
 """
 
 import json
+import logging
+import os
 from datetime import timedelta
-from typing import Optional
+from enum import Enum
+from typing import Dict, List, Optional
 
 import toml
 import yaml
 from cachelib.base import BaseCache
 from cachelib.file import FileSystemCache
-from pydantic import BaseSettings
-from sqlmodel import Session, delete, select
 
-from djqs.exceptions import DJException
-from djqs.models.catalog import Catalog, CatalogEngines
-from djqs.models.engine import Engine
+from djqs.exceptions import DJUnknownCatalog, DJUnknownEngine
 
+_logger = logging.getLogger(__name__)
 
-class Settings(BaseSettings):  # pylint: disable=too-few-public-methods
+
+class EngineType(Enum):
     """
-    Configuration for the query service
+    Supported engine types
     """
 
-    name: str = "DJQS"
-    description: str = "A DataJunction Query Service"
-    url: str = "http://localhost:8001/"
+    DUCKDB = "duckdb"
```

Comment:
Should Spark be included here as well?

Reply:
From what I remember, we removed spark from here once we got rid of the spark session container in the OSS setup. The idea being that if someone extends this open source implementation to include spark support then they can add it to this enum for their implementation.